libraries/src/Installer/InstallerScriptTrait.php

```diff
@@ -0,0 +1,362 @@
+
+    /**
+     * Minimum PHP version required to install the extension
+     *
+     * @var    string
+     * @since  __DEPLOY_VERSION__
+     */
+    protected $minimumPhp;
+
+    /**
+     * Minimum Joomla! version required to install the extension
+     *
+     * @var    string
+     * @since  __DEPLOY_VERSION__
+     */
+    protected $minimumJoomla;
+
+    /**
+     * Allow downgrades of your extension
+     *
+     * Use at your own risk as if there is a change in functionality people may wish to downgrade.
+     *
+     * @var    boolean
+     * @since  __DEPLOY_VERSION__
+     */
+    protected $allowDowngrades = false;
+
+    /**
+     * A list of files to be deleted
+     *
+     * @var    array
+     * @since  __DEPLOY_VERSION__
+     */
+    protected $deleteFiles = [];
+
+    /**
+     * A list of folders to be deleted
+     *
+     * @var    array
+     * @since  __DEPLOY_VERSION__
+     */
+    protected $deleteFolders = [];
+
+    /**
+     * The application object
+     *
+     * @var    CMSApplicationInterface
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    private $application;
+
+    /**
+     * Function called after the extension is installed.
+     *
+     * @param   InstallerAdapter  $adapter  The adapter calling this method
+     *
+     * @return  boolean  True on success
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function install(InstallerAdapter $adapter): bool
+    {
+        return true;
+    }
+
+    /**
+     * Function called after the extension is updated.
+     *
+     * @param   InstallerAdapter  $adapter  The adapter calling this method
+     *
+     * @return  boolean  True on success
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function update(InstallerAdapter $adapter): bool
+    {
+        return true;
+    }
+
+    /**
+     * Function called after the extension is uninstalled.
+     *
+     * @param   InstallerAdapter  $adapter  The adapter calling this method
+     *
+     * @return  boolean  True on success
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function uninstall(InstallerAdapter $adapter): bool
+    {
+        return true;
+    }
+
+    /**
+     * Function called before extension installation/update/removal procedure commences.
+     *
+     * @param   string            $type     The type of change (install or discover_install, update, uninstall)
+     * @param   InstallerAdapter  $adapter  The adapter calling this method
+     *
+     * @return  boolean  True on success
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function preflight(string $type, InstallerAdapter $adapter): bool
+    {
+        $this->extension = $adapter->getName();
+
+        if (!$this->checkCompatibility($type, $adapter)) {
+            return false;
+        }
+
+        if (!$this->checkDowngrade($type, $adapter)) {
+            return false;
+        }
+
+        return $this->customPreflight($type, $adapter);
```

Comment:
I wouldn't do this at all. I would suggest to use trait method rename function instead.

https://3v4l.org/UuJal

```suggestion
        return true;
```